baselines/flwr_baselines/publications/wav2vec2/tedlium/main.py

```diff
@@ -0,0 +1,170 @@
+"""Main module to run the Wav2Vec2.0 TED-LIUM 3 baseline."""
+
+
+import gc
+from argparse import ArgumentParser
+from typing import Callable, Dict
+
+import flwr as fl
+import torch
+from flwr.common import Scalar
+
+import flwr_baselines.publications.wav2vec2.tedlium.strategy as strategy
+from flwr_baselines.publications.wav2vec2.tedlium.client import SpeechBrainClient
+from flwr_baselines.publications.wav2vec2.tedlium.model import model
+
+
+def get_on_fit_config_fn() -> Callable[[int], Dict[str, str]]:
+    """Return a function which returns training configurations."""
+
+    def fit_config(rnd: int) -> Dict[str, str]:
+        """Return a configuration with static batch size and (local) epochs."""
+        config = {"epoch_global": str(rnd), "epochs": str(args.local_epochs)}
+        return config
+
+    return fit_config
+
+
+def evaluate_fn(
+    server_round: int, weights: fl.common.NDArrays, config: Dict[str, Scalar]
+):
+    """Function for centralized evaluation."""
+    _ = (server_round, config)
+    # int model
+    asr_brain, dataset = model.int_model(
+        19999,
+        args.config_path,
+        args.output,
+        args.data_path,
+        args.label_path,
+        args.running_type,
+        args.parallel_backend,
+        evaluate=True,
+    )
+
+    client = SpeechBrainClient("19999", asr_brain, dataset)
+
+    _, lss, err = client.evaluate_train_speech_recogniser(
+        server_params=weights,
+        epochs=1,
+    )
+
+    del client, asr_brain, dataset
+    torch.cuda.empty_cache()
+    gc.collect()
+    return lss, {"Error rate": err}
+
+
+def _parse_arguments():
+    parser = ArgumentParser()
+    parser.add_argument("--data_path", type=str, default="./data/", help="dataset path")
+    parser.add_argument(
+        "--output", type=str, default="./docs/results/fl_fusion/", help="output folder"
+    )
+    parser.add_argument(
+        "--pre_train_model_path",
+        type=str,
+        default=None,
+        help="path for pre-trained starting point",
+    )
+    parser.add_argument(
+        "--label_path",
+        type=str,
+        default="./docs/material/label_encoder.txt",
+        help="path for label encoder file if want to ensure the same encode for every client",
+    )
+    parser.add_argument(
+        "--config_path",
+        type=str,
+        default="./docs/configs/w2v2.yaml",
+        help="path to yaml file",
+    )
+    parser.add_argument(
+        "--running_type", type=str, default="cpu", help="running type of FL "
+    )
+    parser.add_argument(
+        "--min_fit_clients", type=int, default=10, help="minimum fit clients"
+    )
+    parser.add_argument(
+        "--fraction_fit",
+        type=int,
+        default=1,
```

Comment:
Hi, 

I would like to highlight my error I made here. It should be `type=float` and `default=0.01`

Reply:
I made this fix